src/lib.rs

```diff
@@ -41,6 +42,16 @@ pub struct Header {
     pub extra: Map<String, Value>,
 }
 
+impl TryFrom<&[u8]> for Header {
```

Comment:
I don't want to expose this `impl TryFrom`. Please make this a private function/method.

Reply:
Why? Are there any concerns?

Reply:
It's a bit weird to have `TryFrom<&[u8]>` when we are actually expecting a text format (JSON). And it's not implemented for `Claims` and `HeadersAndClaims`, making the API surface a bit inconsistent.

Reply:
> It's a bit weird to have TryFrom<&[u8]> when we are actually expecting a text format (JSON).

How about `TryFrom<&str> `.

> And it's not implemented for `Claims` and `HeadersAndClaims`, making the API surface a bit inconsistent.

We can add the same to `Claims` and `HeadersAndClaims`.